modules/luci-mod-network/htdocs/luci-static/resources/view/network/dhcp.js

```diff
@@ -647,6 +647,23 @@ return view.extend({
 		);
 		o.placeholder = '/etc/dnsmasq.servers';
 
+		s.taboption('forward', form.Flag, 'add_mac',
+			_('Add requestor MAC address to forwards'),
+			_('Add the MAC address of the requestor to DNS queries which are forwarded upstream.'));
+
+		s.taboption('forward', form.Flag, 'strip_mac',
+			_('Remove MAC address before forwarding query'),
+			_('Remove any MAC address information already in downstream queries before forwarding upstream.'));
+
+		o = s.taboption('forward', form.Value, 'add_subnet',
+			_('Add subnet address to forwards'),
+			_('Add a subnet address to the DNS queries which are forwarded upstream, e.g. {32,128} for full requestor address.'));
+		o.optional = true;
+
+		s.taboption('forward', form.Flag, 'strip_subnet',
```

Comment:
This option is mutex with `add_subnet` - so *every option here* needs either) some error checking is necessary or) `depends` to hide parameters. e.g. checking strip subnet hides add subnet and so on.

Reply:
No, all options are independent from each other, see dnsmasq manpage for details.

Reply:
I did 😎 the point being , what’s the use of checking both to add a Mac and remove one.

Reply:
It is possible to pass --add-mac and --strip-mac at the same time and you are right, it looks like this does not make sense. In my understanding, it will remove any MAC address information that is included in the incoming DNS request and then add new MAC information of the direct requestor to the upstream request. This might be important if the sender of the DNS request is e.g. pi-hole. Then the MAc address in the request and the requestor might be different.

Reply:
Nope. I've read it again. They're mutually exclusive.

Reply:
@systemcrash
They are not exclusive. If BOTH the add and strip options are set, incoming EDNS0 options are REPLACED.
https://lists.thekelleys.org.uk/pipermail/dnsmasq-discuss/2022q1/016043.html

Reply:
Finally. Clarity. Yeah I dug into the C code and that lines up.

This will need to be made clear in these options also. Either via a dummy option or under each option.